packages/deployments/contracts/contracts/core/connext/facets/ProposedOwnableFacet.sol

```diff
@@ -245,12 +264,61 @@ contract ProposedOwnableFacet is BaseConnextFacet, IProposedOwnable {
     _setOwner(s._proposed);
   }
 
-  function pause() public onlyOwner {
+  /**
+   * @notice Use to assign an address Router role
+   * Address with Router has permission to add new router
+   * and assign addresses as Role.Router
+   * Can only be called by Owner or Role.Router
+   * @dev requested address will be whitelisted as Role.Router under roles mapping
+   * @param _router - The address to be assigned as Role.Router under roles
+   */
+  function assignRoleRouter(address _router) public onlyOwnerOrRouter {
+    // Use contract as source of truth
+    // Will fail if candidate is already added
+    if (s.roles[_router] == Role.Router) revert ProposedOwnableFacet__assignRoleRouter_alreadyAdded();
+
+    s.roles[_router] = Role.Router;
+    emit AssignRouterRole(_router);
+  }
+
+  /**
+   * @notice Use to assign an address Watcher role
+   * Address with Watcher role has permission to pause
+   * Can only be called by Owner or Role.Admin
+   * @dev requested address will be whitelisted as Role.Watcher under roles mapping
+   * @param _watcher - The address to be assigned as Role.Watcher under roles
+   */
+  function assignRoleWatcher(address _watcher) public onlyOwnerOrAdmin {
+    // Use contract as source of truth
+    // Will fail if candidate is already added
```

Comment:
We don't need the functions to renounce or remove roles, once assign roles to watcher or router, we can't remove role from them.

Reply:
Was this comment intended for a different line? This function seems to be adding the role of watcher.

Regardless, with this setup, an owner *could* "renounce" or "remove" a role simply by assigning something else. For example, an owner could assign the role of router to an EOA with `assignRoleRouter(EOA)` and then later call `assignRoleWatcher(EOA)`. This would make the EOA a watcher instead of a router, effectively revoking their router permissions (even if it's for a diff permission set, it could have destructive effects).

You could fix this by changing the checks at the beginning of these assign methods to
```solidity
if (s.roles[_watcher] != address(0)) revert ProposedOwnableFacet__assignRoleWatcher_alreadyAdded();
```

but there's another problem with that - if we do it that way, and there's no way to *actually* revoke the role, then an EOA will permanently be made a router or watcher (you can't change it back).

Reply:
hmm, we should definitely have a function revoke the added role. Will add a function for it.

Also, the above-shared check is correct, will fix it in the code.